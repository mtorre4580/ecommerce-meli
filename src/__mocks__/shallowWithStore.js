import { shallow } from 'enzyme';

/**
 *  Mock for shallow enzyme add store fake...
 */
const shallowWithStore = (component, store) => {
  const context = {
    store
  };
  return shallow(component, { context });
};

export default shallowWithStore;