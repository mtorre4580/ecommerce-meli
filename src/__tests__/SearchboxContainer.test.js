import React from 'react';
import { createMockStore } from 'redux-test-utils';
import Searchbox from '../containers/SearchboxContainer';
import shallowWithStore from '../__mocks__/shallowWithStore';
import toJson from 'enzyme-to-json';

describe('SearchboxContainer', () => {

    let store;
    let state;

    beforeAll(() => {
        state = {
            form: {
               
            }
        }   
        store = createMockStore(state)
      });

    it('should render success', () => {
      const container = shallowWithStore(<Searchbox />, store);
      expect(container).toBeTruthy();
    });

    it('should render snapshot', () => {
        const container = shallowWithStore(<Searchbox />, store);
        expect(toJson(container)).toMatchSnapshot();
    });

});

