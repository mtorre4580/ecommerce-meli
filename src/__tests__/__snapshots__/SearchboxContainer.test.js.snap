// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchboxContainer should render snapshot 1`] = `
<Route>
  <Component />
</Route>
`;
